### PR TITLE
Allow using name in mapping rather than 'name' label

### LINKED ISSUE #90: metric name should be split from labels in yaml mapping format
#66 was recently merged and I'm a huge fan of the new yaml mappings, however I think it is probably a bad idea to have the `name` label have magic behavior in setting the metric name in the exporter. I /think/ it probably makes more sense to have a `metric_name` parameter one level higher in the mapper config. This should disambiguate the result of the mapping AND allow for, if desired, a mapping which sets a label called `name`.

Proposed change...
this:
```
mappings:
- match: test.case.*
  labels:
    name: "test_case_total"
    blargle: "$1"
```
Would become this:
```
mappings:
- match: test.case.*
  metric_name: "test_case_total"
  labels:
    blargle: "$1"
```

### PR DESCRIPTION
**Pull request by bakins**
fixes #90

I had to modify the tests a bit for this to work as expected. I'm not 100% that the tests are correct.

### CODE REVIEW

**Comment by matthiasr** [comment]
I just realized that there has never been a release with the YAML format. In that case, I would rather remove the special handling of the "name" label altogether.

**Comment by bakins** [comment]
Sgtm. I can remove the name label special handling and update the readme. That sound okay?

**Comment by matthiasr** [comment]
That would be wonderful!

File: README.md [hunk 1]
```diff
@@ -127,15 +127,16 @@ follows:
     test.web-server.foo.bar
      => test_web_server_foo_bar{}
 
+Each mapping in the configutaiton file must define a `name` for the metric.
```
Comment (matthiasr) [inline README.md]: typo: configutation

**Comment by matthiasr** [comment]
Also still needs a rebase, apparently. Please rebase against current master instead of merging master into this branch, I find the resulting history more useful.

**Comment by bakins** [comment]
@matthiasr this wasn't ready to push again - I pushed too soon - and now my branch is in an interesting state. I will have to revisit later when I have time to work through conflicts and merges.

**Comment by matthiasr** [comment]
Alles klar! Let me know if you need help. FWIW – I would squash the whole thing into a single commit, since it's a single well-delineated change. That should also be easier than trying to restore individual commits.

**Comment by bakins** [comment]
@matthiasr I unexpectedly had a few minutes so I have a single commit now based on current master. Thanks!

**Comment by matthiasr** [comment]
Thank you!

**Pull request merged**